spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/partitioner/CassandraPartitionGenerator.scala

```diff
@@ -50,10 +50,20 @@ private[connector] class CassandraPartitionGenerator[V, T <: Token[V]](
   }
 
   private def describeRing: Seq[TokenRange] = {
-    connector.withClusterDo { cluster =>
+    val ranges = connector.withClusterDo { cluster =>
       val metadata = cluster.getMetadata
       for (tr <- metadata.getTokenRanges.toSeq) yield tokenRange(tr, metadata)
     }
+
+    /**
+      * When we have a single Spark Partition use a single global range. This
+      * will let us more easily deal with Partition Key equals and In clauses
+      */
+    if (splitCount == Some(1)) {
+      Seq(ranges(0).copy[V, T](tokenFactory.minToken, tokenFactory.maxToken))
```

Comment:
Do I understand correctly that resulting range has invalid `replicas` and `dataSize`? If so we could skip creating `ranges` and create artificial `TokenRange` on our own.